src/main/java/net/tropicraft/core/common/block/experimental/TropicraftExperimentalLeaveBlock.java

```diff
@@ -0,0 +1,117 @@
+package net.tropicraft.core.common.block.experimental;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.LeavesBlock;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.block.state.properties.IntegerProperty;
+
+import java.util.Random;
+
+public class TropicraftExperimentalLeaveBlock extends LeavesBlock {
+
+    public static IntegerProperty EXTRADISTANCE;
```

Comment:
Yes, some of the more widely shaped ones are pretty big due to the amount of diagonale blocks and the huge palm tree for sure needs it. Ill double check this one later to just confirm what I remember.

Reply:
We could set the MAX_DISTANCE value for each block to shave off even more blockstates since:

- Palm trees max distance it could be away from a log would be 11 (12 as the decay distance)
- Kapok (Rainforest trees / Mahogany) distance would be more like 20 (21 as the decay distance) 
- Mahogany(From the home tree which still is weird) distance would be more like 18 (19 as the decay distance)

I don't think papaya trees need this based on the trees leaves shape

Reply:
Well I had to do some cursed f***ing things for this to work ngl but here is the cursed code:

``` java
public class TropicraftExperimentalLeaveBlock extends LeavesBlock {

    private final int maxDistanceBeforeDecay;

    public IntegerProperty CUSTOM_DISTANCE;

    //Property to check if there has even been a block update to prevent unwanted decay instantly
    public static BooleanProperty TEMPORARY_IGNORE_DECAY = BooleanProperty.create("temp_ignore_decay");

    public TropicraftExperimentalLeaveBlock(Properties settings, int maxDistance){
        super(settings);

        //Cursedness inbound
        this.CUSTOM_DISTANCE = IntegerProperty.create("custom_distance", 1, maxDistance);
        this.maxDistanceBeforeDecay = maxDistance;

        StateDefinition.Builder<Block, BlockState> builder = new StateDefinition.Builder<>(this);

        this.createBlockStateDefinition(builder);
        builder.add(CUSTOM_DISTANCE);

        ObfuscationReflectionHelper.setPrivateValue(Block.class, this, builder.create(Block::defaultBlockState, BlockState::new), "f_49792_");

        this.registerDefaultState(this.stateDefinition.any()
                .setValue(PERSISTENT, false)
                .setValue(CUSTOM_DISTANCE, maxDistanceBeforeDecay)
                .setValue(TEMPORARY_IGNORE_DECAY, false));
    }
...
}
```